src/core/types/addons.js

```diff
@@ -156,6 +156,17 @@ export type ExternalAddonType = {|
   weekly_downloads: number,
 |};
 
+export type PlatformFilesType = {|
+  // This seems necessary to help Flow know that computed
+  // keys always return an AddonFileType.
+  [anyPlatform: string]: ?AddonFileType,
```

Comment:
It looks like you can delete this now. I think it was only happening before because of how the platform files object was created by assignment in a loop.